### DISCUSSION TITLE
Best practices to hack new Yi camera?

### DISCUSSION THREAD

**Discussion by chrbar11** [roleoroleo/yi-hack-Allwinner-v2#346] [General]
Hi,

I've 4 new Yi 1080p Home (IFUS 9.0.36*) from a Family Pack.
I've run y211ga hack on one camera, and everything run great.
Thanks for your nice hack!

I've some questions before hack my 3 other camera.

Is it recommanded to update camera to the last Yi firmware version (9.0.36*) before apply the hack ?

Is it recommanded to hack the camera without/before register it to Yi App (YI Home) ?
I had to join my first camera to know the native Yi firmware version.

Could you please tell me what is involved in registering the camera on the Yi app ?
Yi obtains my current IP address, and my camera serial number... anything else... something not secure for me?

After set Authentication into YI-HACK Configurations, my camera was no more reachable from Yi App (YI Home).
It's perfect, it's what I want!
I also have Disable Cloud.
Are there other settings to cut all links with YI services from Internet ?

Can I duplicate and use the same SD content (including personal settings) for many Yi 1080p Home cameras (from the same pack, natively built with Yi firmware version) ?

Thank you a lot for your help and advices.
Chris

**Comment by alexfranke** [comment]
One of my 4-pack has a slight defect, so I'm returning it soon -- let me know if there's anything in particular you'd like me to try out with it. 

Meantime, I took the opportunity to experiment a bit. It originally had firmware 9.0.36.00_202011101611 with Yi Hack 0.1.9 installed and it worked great. Since then, this is what I tried: 
 * Upgraded Yi Hack to 0.2.0 -- I didn't to any exhaustive testing, but I get the website, and MQTT seems to work fine. 
 * Upgraded to current YI Firmware (9.0.36.02_202105081647), same SD card as before. Camera didn't appear to accessible from either the Yi app or the Yi Hack http server. 
 * Removed SD card, rebooted camera. Camera didn't appear to be accessible from Yi app (it looked like it hung after 25% "connected") although in hindsight I didn't give it much time to complete. Later in this process it also hung for a little bit at 25%. 
 * Deleted contents of SD card, installed the Yi Hack 0.2.0 files fresh (as if I was just installing the hack for the first time). The camera didn't give me the flashing blue light I was expecting between 30 seconds and 2 minutes after restarting. I tried the camera's http server, which appeared to load only the header and nothing else. After a couple of minutes, I refreshed, and it loaded everything and looks like normal. It was also accessible from the Yi app, after appearing to hang for a bit at 25%. 
 * Configured and rebooted with latest Yi firmware and latest Yi hack, and it seems to work reasonably well
   - MQTT works, 
   - I can connect to the rtsp server with vlc. 
   - The web server seems to take a bit longer to reload -- often looks like it's hanging after just loading the header, and I don't see the status page unless I click on configurations first. 
   - I noticed that the camera serial number isn't reported properly (it's all zeros -- don't know if this was the case before the camera's firmware update with hack v 0.2

**Reply by roleoroleo** [reply]
> * I noticed that the camera serial number isn't reported properly (it's all zeros -- don't know if this was the case before the camera's firmware update with hack v 0.2

It's a frequent problem. I will check it.

**Comment by roleoroleo** [comment]
> Is it recommanded to update camera to the last Yi firmware version (9.0.36*) before apply the hack ?

The update could overwrite some hacked file.
It's recommended to update the camera.

> 
> Is it recommanded to hack the camera without/before register it to Yi App (YI Home) ? I had to join my first camera to know the native Yi firmware version.

No. Join the cam and then apply the hack.

> Could you please tell me what is involved in registering the camera on the Yi app ? Yi obtains my current IP address, and my camera serial number... anything else... something not secure for me?

Apart from the data of your account, the stream i sent to the yi servers.
But you can avoid this setting the "Disable cloud" option.

> After set Authentication into YI-HACK Configurations, my camera was no more reachable from Yi App (YI Home). It's perfect, it's what I want! I also have Disable Cloud. Are there other settings to cut all links with YI services from Internet ?

It's not reachable because you set "Disable cloud" option.
No other settings.

> Can I duplicate and use the same SD content (including personal settings) for many Yi 1080p Home cameras (from the same pack, natively built with Yi firmware version) ?

Not after setup.
There are settings you can't share: hostname, mqtt client id, etc...
